nienbo/cache-buildkite-plugin
# feat: Adding 'save-cache' property for S3 cache

## What it does?
It makes the plugin saves the tar file on `/tmp` after downloading it for the first time, and re-uses that file on any further execution.

This is to complement the #36 implementation in further executions of builds on the same node, and when the same cache file is used by multiple pipelines.

## How can I test it?
I've created a test to check it works.

Also, use this Plugin from this Fork on a pipeline, and enable the property like this:
```yaml
...

  - plugins:
    - gencer/cache#v2.4.13:
        id: ruby # or ruby-3.0
        backend: s3
        key: "v1-cache-{{ id }}-{{ runner.os }}-{{ checksum 'Gemfile.lock' }}"
        restore-keys:
          - 'v1-cache-{{ id }}-{{ runner.os }}-'
          - 'v1-cache-{{ id }}-'
        s3:
          prefere-local: true
        paths:
          - 'bundle/vendor'
...
```

## Issues

 * #36

## Extra Comments

This is not really to fully address #36 but to complement it.
